### DISCUSSION TITLE
How to import duration details of events to view insights

### DISCUSSION THREAD

**Discussion by daamiian** [YukiGasai/obsidian-google-calendar#107] [Q&A]
I manage my tasks in obsidian, each one is it's own note with an estimated time, due date, etc. in the frontmatter. I use time blocking to schedule time to work on certain projects' tasks as daily sprints. E.g.

**Work Sprint:** time blocked on my calendar in blocks: _09:00 - 12:00_, _13:00 - 15:00 every day_
**Thesis Sprint:** time blocked on my calendar in blocks: _08:00 - 09:00_, _16:00 - 17:00 every other day_
etc.

Some blocks are project specific, some more general, but what I'm doing is displaying all of my task notes in dataview tables and I need to calculate the time available for each task based on the blocked time on my calendar for those projects. 

I need to find a way to calculate the amount of time available that has been time blocked as an event in my google calendar between now and the due date of a task. Somehow I need that accessible to dataview tables. I can also use these durations to show weekly time spent in certain meetings, etc. Any ideas?

**Comment by YukiGasai** [comment]
Hi, this sounds pretty complex. 
This plugin can't help you with finding your blocked time for events, if the event is not inside your google calendar.
If the events can be filtered independent of your notes and is inside the google calendar, you could just use the `getEvents` function from the API to get the data directly and then work with that to calculate your time.
In the example, the combined blocked date of a specific reoccurring event is summed up over the current week. It will be wrong, if the event will overlap with itself.
~~~
```dataviewjs
const {getEvents} = this.app.plugins.plugins["google-calendar"].api;
const startMoment = window.moment().startOf("week");
const endMoment   = window.moment().endOf("week");

//Here you can define the range of the events you need
//More then a month will most likely not work
const events = await getEvents({
startDate: startMoment,
endDate: endMoment,
});

// Here you could filter the events for what you need
const filteredEvents = events.filter((event) => {
	return event.summary?.startsWith("FOUR")
})

const blockedTime = filteredEvents.reduce((sum, event) => {
	if(event.start.date){
		return sum += 24;
	}
	const start = window.moment(event.start.dateTime);
	const end   = window.moment(event.end.dateTime)
	return sum += end.diff(start, 'minutes', true);

}, 0)

console.log(blockedTime);
```
~~~
